src/apps/Permissions/Permissions.js

```diff
@@ -65,244 +35,177 @@ class Permissions extends React.Component {
       return {
         screen,
         address: data,
-        app: this.getAppByProxyAddress(data),
+        app: getAppByProxyAddress(data, apps),
         secondaryScreen,
         secondaryData,
       }
     }
-
-    if (screen === 'entity' && isAddress(data)) {
-      return { screen, address: data }
-    }
-
-    return home
   }
 
-  getAppByProxyAddress(proxyAddress) {
-    if (!proxyAddress) {
-      return null
-    }
-    return this.props.apps.find(app =>
-      addressesEqual(app.proxyAddress, proxyAddress)
-    )
-  }
-
-  goToHome = () => {
-    this.props.onParamsRequest(null)
-  }
-
-  handleOpenApp = proxyAddress => {
-    this.props.onParamsRequest(`app.${proxyAddress}`)
-  }
-
-  handleOpenEntity = address => {
-    if (this.getAppByProxyAddress(address)) {
-      return this.handleOpenApp(address)
-    }
-    this.props.onParamsRequest(`entity.${address}`)
-  }
-
-  handleManageRole = (proxyAddress, roleBytes) => {
-    this.props.onParamsRequest(`app.${proxyAddress}.role.${roleBytes}`)
-  }
-
-  createPermission = () => {
-    this.setState({ showAssignPermissionPanel: true })
-  }
-
-  closeAssignPermissionPanel = () => {
-    this.setState({ showAssignPermissionPanel: false })
-  }
+  return home
+}
 
-  closeManageRolePanel = () => {
-    const { params, onParamsRequest } = this.props
-    const location = this.getLocation(params)
+function Permissions({
+  apps,
+  appsLoading,
+  onParamsRequest,
+  params,
+  permissionsLoading,
+  wrapper,
+}) {
+  const { layoutName } = useLayout()
+  const [showAssignPermissionPanel, setShowAssignPermissionPanel] = useState(
+    false
+  )
+  const { resolveRole } = usePermissions()
+  const scrollTopElement = useRef(null)
+
+  // `params` should change every time we navigate into and out of a detailed
+  // permissions view, so this ensures the user starts at the top of the screen
+  // on every navigation change
+  useEffect(() => {
+    scrollTopElement.current.scrollIntoView()
```

Comment:
A thing I've wondered is maybe we should not scroll to top when coming back to the list, not sure if we can do this correctly but maybe this can provide a way for the user to come back to the same scrolling position when they chose an item.

Reply:
This would be nice, but maybe a bit tricky? We could remember the scroll position and maybe use a lifecycle hook to know when the rendering is done to scroll back into position?

Reply:
> This would be nice, but maybe a bit tricky

I think browsers give the scroll position back on navigating back, maybe we can figure a way to use this behaviour with a different router, but yeah as you mention tricky and most probably out of scope for the time being.

Reply:
Something else that we've been meaning to do is add the transition back into CardLayout, and when we do so, we could make the detail page an overlay over the CardLayout itself so that we automatically keep the state and scroll position underneath.

Reply:
Yes, that's another approach that works but be mindful of doing that for mobile cause it's expensive to have things on top of things because they still get rendered.

Reply:
Moved to https://github.com/aragon/aragon/issues/926.